ui/src/AppCharting/AppChart.svelte

```diff
@@ -0,0 +1,253 @@
+<script lang="ts">
+  import {errorStore} from '../utility/stores';
+  import LineChart from './LineChart.svelte';
+
+  import {
+    chartDataFactory,
+    classNameFromBenchmarkKey,
+    BY_SIZE,
+    BY_TIMESTAMP,
+    MISSING_VALUE
+  } from './chart-data-extractor';
+
+  import type {ChartFactoryData} from './chart-data-extractor';
+
+  import {
+    axisFactory,
+    getAxisYLabel,
+    getAxisXLabel,
+    getAxisYMin,
+    getAxisYType,
+    getLegendTitle,
+    getAxisXTimeStampType,
+    DEFAULT_CHART_HEIGHT
+  } from './chart-layout-helpers';
+  import type {
+    BenchmarkIdentifier,
+    Benchmarks,
+    FormConfiguration,
+    Scenario,
+    StatProperties,
+    TimestampViewModel
+  } from '../types';
+  import type {Primitive} from 'c3';
+
+  type ErrorEntry = {
+    key: BenchmarkIdentifier;
+    scenario: Scenario;
+    dateTime: string;
+    size: string;
+  };
+
+  //----------------------------------------------------------------
+  // Props
+  //------------------------------------------------------------
+  export let form: FormConfiguration;
+  export let selectedTimestamps: BenchmarkIdentifier[];
+
+  export let benchmarks: Benchmarks = {};
+  export let timestamps: TimestampViewModel[] = [];
+  export let statProperties: StatProperties;
+
+  //----------------------------------------------------------------
+  // Local State
+  //------------------------------------------------------------
+  let errors = new Map<BenchmarkIdentifier, ErrorEntry>();
+  let chartData: ChartFactoryData = {columns: [], x: 'x', names: {}};
+  let axisConfigurations = axisFactory();
+
+  //----------------------------------------------------------------
+  // Computed Properties
+  //------------------------------------------------------------
+  $: scenario = form.scenario;
+  $: chartType = form.chartType;
+
+  $: isReady = benchmarks && statProperties && form && true;
+
+  $: hasNodes = scenario === 'disco' || scenario.startsWith('showtime_');
+  $: legendTitle = getLegendTitle(form, {hasNodes});
+  $: title = [form.scenario, form.plotType, form.statName].join(' \uFF5C ');
+
+  // Axis Configuration
+  $: axisConfigurations[BY_TIMESTAMP].x.type = getAxisXTimeStampType(form);
+  $: axisConfigurations[BY_TIMESTAMP].x.tick.fit = form.useTimeSeries;
+  $: axis = axisConfigurations[chartType];
+
+  // X and Y Label
+  $: if (chartData && axis && isReady) {
+    axis.y.type = getAxisYType(form);
+    axis.y.min = getAxisYMin(form, chartData);
+    if (typeof axis.y.label !== 'string') {
+      axis.y.label.text = getAxisYLabel(form, {statProperties});
+    }
+    if (typeof axis.x.label !== 'string') {
+      axis.x.label.text = getAxisXLabel(form, {hasNodes});
+    }
+  }
+
+  $: if (isReady) {
+    const selected = timestamps.filter(({key}) => {
+      return selectedTimestamps.indexOf(key) !== -1;
+    });
+
+    const factory = chartDataFactory(chartType);
+    factory(benchmarks, selected, form).then(onLoaded).catch(onError);
+  }
+
+  $: {
```

Comment:
A minor thing, but you don't need the braces here and can turn these three lines into one.

Reply:
Fixed 81f09d1